Improve content in Alter workout SQL DDL

kapnobatai137 opened the pull request:
1. Add Column:
- introduced syntax for adding a column that is also a `PRIMARY KEY`
- changed tables to md format

2. Add Constraint:
- introduced syntax for adding a `FOREIGN KEY` constraint
- changed tables to md format
- introduced syntax for dropping a constraint

3. Add Default Value:
- changed title to Alter Characteristics
- introduced alter data type concept
- added example syntax for alter data type
- introduced rules for altering data type
- changed add default value syntax to ANSI Standard
- introduced drop default value concept
- introduced syntax for adding not null clause
- introduced syntax for dropping not null clause
- changed tables to md format
- improved revision question to request adding a new default value and dropping a not null clause

4. Add trigger:
- changed syntax to ANSI Standard
- Added `ON UPDATE` and `ON DELETE` concepts
- added argument concepts
  - `CASCADE`
  - `SET NULL`
  - `SET DEFAULT`
  - `RESTRICT`
  - `NO ACTION`
- modified revision question to fit new syntax
- wrote new practice question
  - 5 correct/10 total answers

5. Add rule:
- insight was removed due to being too advanced and *flavour* specific (PostgreSQL and MySQL Server only)
- it will be added in a future workout that will contain differences between ANSI Standard and each *flavour*

enki-hq commented:
Rule insight:code-width failed for files: 
  - sql/ddl/alter/add-column.md

enki-hq commented:
### Insight previews:
- [ ] [enkidevs/curriculum/.archived/sql/ddl/alter/add-default-value.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/.archived/sql/ddl/alter/add-default-value.md?ref=13c83236e8d1d392a6c3ec3d9392bbb5d9344a6f)
- [ ] [enkidevs/curriculum/.archived/sql/ddl/alter/add-rule.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/.archived/sql/ddl/alter/add-rule.md?ref=13c83236e8d1d392a6c3ec3d9392bbb5d9344a6f)
- [ ] [enkidevs/curriculum/sql/ddl/alter/add-column.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/sql/ddl/alter/add-column.md?ref=13c83236e8d1d392a6c3ec3d9392bbb5d9344a6f)
- [ ] [enkidevs/curriculum/sql/ddl/alter/add-constraint.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/sql/ddl/alter/add-constraint.md?ref=13c83236e8d1d392a6c3ec3d9392bbb5d9344a6f)
- [ ] [enkidevs/curriculum/sql/ddl/alter/add-trigger.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/sql/ddl/alter/add-trigger.md?ref=13c83236e8d1d392a6c3ec3d9392bbb5d9344a6f)
- [ ] [enkidevs/curriculum/sql/ddl/alter/alter-characteristics.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/sql/ddl/alter/alter-characteristics.md?ref=13c83236e8d1d392a6c3ec3d9392bbb5d9344a6f)

kapnobatai137 commented:
Add trigger:
- split into 2 insights

Add Trigger (1):
- create trigger syntax
- `RESTRICT` argument
- `NO ACTION` argument
- fixed practice question to fit new content
- removed revision question
  - no need to have 2 revision questions on this topic

Add Trigger (2):
- `CASCADE` argument
- `SET NULL` argument
- `SET DEFAULT` argument
- modified practice question to fit new content
- slightly improved revision question to contain content from `Add Trigger (1)`

nem035 commented on sql/ddl/alter/add-trigger-2.md:
```diff
@@ -0,0 +1,135 @@
+
+### CASCADE
+
+Using the `CASCADE` argument means that when a row in the parent table is deleted or updated, all the matching rows in the child tables are deleted or updated. Here is an example syntax:
```
Maybe add a "too" at the end of the first sentence?

Using X says that when parent is updated, the child is updated too.

nem035 commented on sql/ddl/alter/add-trigger-2.md:
```diff
@@ -0,0 +1,135 @@
+---
+## Revision
+
+Complete the following syntax such that the *trigger* named 'cstr_2' does the following:
```
Should we say `cstr_2` instead of 'cstr_2'?

nem035 commented on sql/ddl/alter/add-trigger-2.md:
```diff
@@ -0,0 +1,135 @@
+- when deleting from the parent table the corresponding record in the child table is set to its default value
+- when updating the parent table the corresponding record in the child table is updated
+
+prohibits deleting a record from the `pokemons` table if there are any matching records in the `region` table.
```
Is this sentence missing a beginning phrase?

kapnobatai137 replied:
I somehow forgot to completely delete the previous question
